Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndo.cpp

```diff
@@ -107,6 +132,13 @@ namespace AzToolsFramework
                     // Update the cached instance dom corresponding to the entity so that the same modified entity isn't reloaded again.
                     entityPathInDom.Set(cachedDom->get(), AZStd::move(endStateCopy));
                 }
+                // Create a copy of the dom of the end state so that it shares the lifecycle of the cached dom.
+                PrefabDom endStateCopy;
+                endStateCopy.CopyFrom(endState, cachedDom->get().GetAllocator());
+                Prefab::PrefabDomPath entityPathInDom(entityAliasPath.c_str());
+
+                // Update the cached instance dom corresponding to the entity so that the same modified entity isn't reloaded again.
+                entityPathInDom.Set(cachedDom->get(), AZStd::move(endStateCopy));
```

Comment:
Not sure I follow the logic here. This is the same as the code in lines 125 - 134

Reply:
I think after I fixed the duplication, this has been resolved as well? Let me know if it doesn't.